kentik_api_library/kentik_api/api_connection/api_connector.py

```diff
@@ -1,32 +1,50 @@
 # Standard library imports
 import logging
+from http import HTTPStatus
 from typing import Optional, Dict, Any
 
 # Third party imports
 import requests
 
 # Local application imports
-from kentik_api.auth.auth import KentikAuth
 from kentik_api.api_calls.api_call import APICall, APICallMethods
 from kentik_api.api_connection.api_call_response import APICallResponse
+from kentik_api.auth.auth import KentikAuth
+from kentik_api.public.errors import (
+    AuthError,
+    BadRequestError,
+    NotFoundError,
+    RateLimitExceededError,
+    UnavailabilityError,
+    KentikAPIError,
+)
 
 
 class APIConnector:
     """ APIConnector implements APIConnectorProtocol. Allows sending authorized http requests to Kentik API """
 
     DEFAULT_HEADERS = {"Content-Type": "application/json"}
-    BASE_API_US_URL = "https://api.kentik.com/api"
-    BASE_API_EU_URL = "https://api.kentik.eu/api"
 
     def __init__(self, api_url: str, auth_email: str, auth_token: str) -> None:
         self._api_url = api_url
         self._auth = KentikAuth(auth_email, auth_token)
         self._logger = logging.getLogger(__name__)
 
     def send(self, api_call: APICall, payload: Optional[Dict[str, Any]] = None) -> APICallResponse:
-        url = self._get_api_query_url(api_call.url_path)
-        response: requests.Response
+        try:
+            response = self._do_request(api_call, payload)
+        except requests.Timeout as e:
+            raise UnavailabilityError(str(e)) from e
+        except requests.RequestException as e:
+            raise KentikAPIError(str(e)) from e
+
+        self._logger.debug(f"Got API response: {response}")
```

Comment:
can we also log the request so one can have full request-reponse data for analysis in case of failure?

Reply:
Done